src/transactions/CategoryFilter.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { CATEGORIES } from '../data';
+
+function CategoryFilter({ handleFilter }) {
+  function onSelectChanged(event) {
+    const { value } = event.target;
+
+    handleFilter({ categoryId: value });
+  }
+
+  function onClear() {
+    handleFilter({ categoryId: null });
+  }
+
+  return (
+    <div className="row">
+      <div className="col col-3 mr-2">
+        <select className="form-select" name="categoryFilter" onChange={onSelectChanged}>
+          <option>Filter by Category</option>
+          {CATEGORIES.map((category) => (
```

Comment:
We actually need to get this data from Redux, not from the data.js file. That data file is just used to give Redux some initial data when it first starts up for convenience while we're developing, since we don't have a backend and database yet. The actual categories are stored in Redux and that's what will stay updated as the user adds, deletes etc.